tuna/mituna_interface.py

```diff
@@ -122,3 +124,83 @@ def execute_docker(self, worker, docker_cmd, machine):
     #  self.logger.info(line.rstrip())
 
     return True
+
+  def add_tables(self):
+    """Add library specific tables"""
+    return self.add_tables()
+
+  def determine_num_procs(self, machine):
+    """Determine number of processes by compute capacity"""
+    worker_ids = None
+    env = get_env_vars()
+    if env['slurm_cpus'] > 0:
+      num_procs = int(env['slurm_cpus'])
+    else:
+      # JD: This sould be the responsibility of the machine class
+      num_procs = int(machine.get_num_cpus() * .6)
+
+    worker_ids = range(num_procs)
+
+    if len(worker_ids) == 0:
+      self.logger.error('num_procs must be bigger than zero to launch worker')
+      self.logger.error('Cannot launch worker on machine: %s', machine.id)
+      worker_ids = None
+
+    return worker_ids
+
+  def get_f_vals(self, machine, worker_ids, args):
+    """Determine kwargs for worker_interface"""
+    f_vals = self.compose_f_vals(machine)
+    f_vals["num_procs"] = Value('i', len(worker_ids))
+    f_vals['envmt'] = self.get_envmt(args)
+    return f_vals
+
+  def get_envmt(self, args):
+    """Get runtime envmt"""
+    raise NotImplementedError("Not implemented")
+
+  def compose_f_vals(self, machine):
+    """! Compose dict for WorkerInterface constructor
+      @param args The command line arguments
+      @param machine Machine instance
+    """
+    f_vals = {}
+    f_vals["barred"] = Value('i', 0)
+    f_vals["bar_lock"] = Lock()
+    #multiprocess queue for jobs, shared on machine
+    f_vals["job_queue"] = mpQueue()
+    f_vals["job_queue_lock"] = Lock()
+    f_vals["result_queue"] = mpQueue()
+    f_vals["result_queue_lock"] = Lock()
+    f_vals["machine"] = machine
+    f_vals["b_first"] = True
+    f_vals["end_jobs"] = Value('i', 0)
```

Comment:
As discussed, this could be removed and the objects which have these attributes be passed instead.

Reply:
This is a big change. Can i make this a separate issue and address at a different time? There have already been many parallel updates to source code improvement in general, unrelated to adding a new lib.